automation/tables.py

```diff
@@ -1,15 +1,126 @@
-from . import models
-from .functions import inbound, members
-from .clients import airtable
-
-INBOUND = airtable.TableSpec(
-    name="inbound",
-    model_cls=models.InboundModel,
-    status_to_cb={None: inbound.NewCallback},
-)
+import abc
+from functools import cached_property
+
+import sendgrid
+
+from . import config, models
+from .functions import delivery, inbound, members
+from .clients import airtable, slack
+
+
+class PollableTable(abc.ABC):
+
+    table_spec = ...
+
+    def __init__(self, conf: config.Config, read_only=False):
+        self.conf = conf
+        self.read_only = read_only
+
+    @classmethod
+    def get_airtable(cls, conf: config.AirtableConfig, read_only=False):
+        return cls.table_spec.get_airtable_client(conf, read_only=read_only)
+
+    def poll_table(self):
+        client = self.get_airtable(self.conf.airtable, self.read_only)
+        return client.poll_table(self.on_status_update)
+
+    @abc.abstractmethod
+    def on_status_update(self, record):
+        ...
+
+
+class SlackMixin:
```

Comment:
If your `PollableTable`'s callback method needs slack or email clients, add these mixins and you can use their properties in your callback method. They'll be cached with a scope that's the lifetime of your `PollableTable` instance.